src/provisioner/provisioner.py

```diff
@@ -227,11 +227,10 @@ def _provision_components(cortx_conf: MappedConf, mini_prov_interfaces: list, ap
                     raise CortxProvisionerError(
                         rc, "%s phase of %s, failed. %s", interface,
                         component_name, err)
-                # Update version for each component if upgrade successful.
-                if apply_phase == PROVISIONING_STAGES.UPGRADE.value:
-                    component_version = CortxProvisioner.cortx_release.get_component_version(
-                        component_name)
-                    cortx_conf.set(f'{key_prefix}>version', component_version)
+                # Update version for each component if Provisioning successful.
+                component_version = CortxProvisioner.cortx_release.get_component_version(
```

Comment:
Note: mini provisioner interfaces are idempotent. So its to run them twice but not okay to skip. We should record new version only after all the interfaces have been successfully completed ?

Reply:
done